Added nginx reverse proxy example to GettingStarted.md

iamed18 opened the pull request:
This PR comes after a short conversation in https://github.com/jupyterhub/jupyterhub/issues/509 and over at [the Google group](https://groups.google.com/forum/#!topic/jupyter/tPcQKr8bznY) about getting JupyterHub running behind an nginx reverse proxy.

The only changes in this PR are to Getting Started.md under the networking section.

willingc commented:
Thanks @iamed18 for the PR related to #509. I'm glad you wrote this up, and I'll be happy to review it.

willingc commented on docs/source/getting-started.md:
```diff
@@ -462,6 +464,120 @@ export CONFIGPROXY_AUTH_TOKEN=super-secret
+	ssl_certificate_key /etc/letsencrypt/live/HUB.DOMAIN.TLD/privkey.pem
+	
+	ssl_protocols TLSv1 TLSv1.1 TLSv1.2;
+    ssl_prefer_server_ciphers on;
```
Any reason why the indentation changes here?

iamed18 replied:
The primary reason here is my own carelessness. I'll clean that up.

willingc commented on docs/source/getting-started.md:
```diff
@@ -462,6 +464,120 @@ export CONFIGPROXY_AUTH_TOKEN=super-secret
+		try_files $uri $uri/ =404;
+	}
+
+	# Managing requests to verify letsencrypt host
```
Indentation check

iamed18 replied:
Could you elaborate on this one? I don't see anything out of place at first glance.

willingc commented on docs/source/getting-started.md:
```diff
@@ -462,6 +464,120 @@ export CONFIGPROXY_AUTH_TOKEN=super-secret
+	listen 443;
+	ssl on;
+	
+	# INSERT OTHER SSL PARAMETERS HERE AS ABOVE
```
Perhaps make this more prominent

```
# ---------------------------------------------------------------------------
# INSERT OTHER SSL PARAMETERS HERE AS ABOVE
# ---------------------------------------------------------------------------
```

iamed18 replied:
That's a good idea. Will do.

willingc commented on docs/source/getting-started.md:
```diff
@@ -462,6 +464,120 @@ export CONFIGPROXY_AUTH_TOKEN=super-secret
+* JupyterHub is running on a server, accessed *only* via `HUB.DOMAIN.TLD:443`
+* On the same machine, `NO_HUB.DOMAIN.TLD` strictly serves different content, also on port `443`
+* `nginx` is used to manage the web servers / reverse proxy (which means that only nginx will be able to bind two servers to `443`)
+* After testing, the server in question should be able to score an A+ on the Qualys SSL Labs [SSL Server Test](https://www.ssllabs.com/ssltest/)
```
Nice! 
Missing period at end of sentence.

iamed18 replied:
Will fix.

willingc commented:
Hi @iamed18, This looks really good. A couple of suggestions:

- Let's break out the examples to their own page ``config-examples.md`` and add it to the Table of Contents in the Configuration section
- Some small suggestions on spacing

iamed18 commented:
That sounds good @willingc. I'll try to mock something up in the next few days.

willingc commented:
@iamed18 Sounds great! Thanks.

As for the indentation, looking at the richdiff it looks fine to me.

willingc commented:
Hi @iamed18! I wanted to know if you are still finding time to work on this or if you would prefer that we finish the changes. Either one is cool. I just would like to see the content that you provided find its way into the documentation for others to use. 😄

iamed18 commented:
@willingc I do certainly plan on getting to it. I've just been a busy bee attempting to migrate my existing installation of jupyterhub from Debian to Scientific Linux. I will get to this soon-ish!

iamed18 commented:
@willingc  Okay I'm finally back. I've moved the config examples to their own file, but I don't know how to add the new item to the ToC because I can't actually find the ToC. I added a link to the bottom of getting-started.md, but I'm not sure that's what you were after.

willingc commented:
@iamed18 Thanks a bunch for finishing up this PR. I've merged this and appreciate the contribution. :cake: :tada:

The ToC is a bit tricky. It's in ``index.rst`` but if you look at it in GitHub's Markdown Viewer it doesn't show the ToC entries. You would need to click the "raw file" button to see the actual ToC items.

iamed18 commented:
Okay. Thanks for that clarification. Should I do that and submit another PR
or is that something someone on your team will do now?

On Thu, Jul 28, 2016 at 4:14 PM Carol Willing <notifications@github.com>
wrote:

> @iamed18 <https://github.com/iamed18> Thanks a bunch for finishing up
> this PR. I've merged this and appreciate the contribution. 🍰 🎉
>
> The ToC is a bit tricky. It's in index.rst but if you look at it in
> GitHub's Markdown Viewer it doesn't show the ToC entries. You would need to
> click the "raw file" button to see the actual ToC items.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/jupyterhub/jupyterhub/pull/602#issuecomment-236027303>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AIIBfB8xHIZO2oHX72EQUtjiRsXXVAO_ks5qaRu1gaJpZM4Ivg-I>
> .
>

willingc commented:
@iamed18 I just wrote a PR for it since I didn't wish to bother you with it. You did the hard stuff 👍 

I just wanted you to know for future reference re: ToC since I'm hoping to see future contributions as you come across things that would be helpful for others 😄 

Thanks again!
